File: backend/java/api_gateway/src/main/java/com/ticketflow/api_gateway/proxy/profile/ProfilesApiProxy.java
package com.ticketflow.api_gateway.proxy.profile;

import com.ticketflow.api_gateway.models.exceptions.NotFoundException;
import com.ticketflow.api_gateway.models.profile_service.Profile;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ProfilesApiProxy {

    private ProfilesApiFeignClient profilesFeignClient;

    @Autowired
    public ProfilesApiProxy(ProfilesApiFeignClient profilesFeignClient) {
        this.profilesFeignClient = profilesFeignClient;
    }

    public Profile getByUserEmail(String userEmail) throws NotFoundException {
        return profilesFeignClient.getByUserEmail(userEmail).getBody();
    }

    public void add(Profile profile) {
        profilesFeignClient.add(profile).getBody();
    }
}

File: backend/dotnet/ticketflow/TicketFlow.ApiGateway/Proxy/ProfilesApiProxy.cs
using System.Net.Http;
using TicketFlow.ApiGateway.Models.ProfileService;

namespace TicketFlow.ApiGateway.Proxy
{
    internal class ProfilesApiProxy : IProfilesApiProxy
    {
        private readonly IHttpClientFactory httpClientFactory;

        public ProfilesApiProxy(IHttpClientFactory httpClientFactory)
        {
            this.httpClientFactory = httpClientFactory;
        }

        public Profile GetById(int id)
        {
            throw new System.NotImplementedException();
        }

        public Profile GetByUserAsync(string userEmail)
        {
            throw new System.NotImplementedException();
        }

        public Profile Add(Profile profile)
        {
            throw new System.NotImplementedException();
        }
    }
}